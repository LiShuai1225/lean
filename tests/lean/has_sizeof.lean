set_option pp.binder_types true

namespace X0

inductive foo (A : Type) : Type
| mk : A -> foo -> foo

check @foo.has_sizeof
check @foo.mk.has_sizeof_spec

end X0

namespace X1

inductive foo (A : Type) (B : nat -> A -> A -> Type) (C : Type)  : Type
| mk : Pi (a : A) (n : nat), B n a a -> (bool -> B n a a) -> (bool -> C) -> (bool -> foo) -> foo

check @foo.has_sizeof
check @foo.mk.has_sizeof_spec

end X1
