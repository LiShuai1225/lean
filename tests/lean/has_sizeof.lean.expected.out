foo.has_sizeof : Π (A : Type) [A_inst : has_sizeof A], has_sizeof (foo A)
foo.mk.has_sizeof_spec :
  ∀ (A : Type) [A_inst : has_sizeof A] (a : A) (a_1 : foo A),
    has_sizeof.sizeof (foo.mk a a_1) = 1 + has_sizeof.sizeof a + has_sizeof.sizeof a_1
foo.has_sizeof :
  Π (A : Type) (B : ℕ → A → A → Type) (C : Type) [A_inst : has_sizeof A]
  [B_inst : Π (a : ℕ) (a_1 a_2 : A), has_sizeof (B a a_1 a_2)], has_sizeof (foo A B C)
foo.mk.has_sizeof_spec :
  ∀ (A : Type) (B : ℕ → A → A → Type) (C : Type) [A_inst : has_sizeof A]
  [B_inst : Π (a : ℕ) (a_1 a_2 : A), has_sizeof (B a a_1 a_2)] (a : A) (n : ℕ) (a_1 : B n a a)
  (a_2 : bool → B n a a) (a_3 : bool → C) (a_4 : bool → foo A B C),
    has_sizeof.sizeof (foo.mk a n a_1 a_2 a_3 a_4) = 1 + has_sizeof.sizeof a + has_sizeof.sizeof n + has_sizeof.sizeof
      a_1 + has_sizeof.sizeof a_2 + has_sizeof.sizeof a_3
